### PR TITLE
pass-otp: 1.0.0 -> 1.1.0

### PR DESCRIPTION
**Pull request by tadfisher**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/security/pass-otp/default.nix [hunk 1]
```diff
@@ -5,8 +5,8 @@ stdenv.mkDerivation {
   src = fetchFromGitHub {
     owner = "tadfisher";
     repo = "pass-otp";
-    rev = "f2feb3082324a91089782af9b7fbb71d34aa213d";
-    sha256 = "0iklvcfgw1320dggdr02lq3bc7xvnd2934l1w9kkjpbsfmhs955c";
+    rev = "9edb479405156f5712dd101661b86bd599904e8d";
```
Comment (Mic92) [inline pkgs/tools/security/pass-otp/default.nix]: Please release tags here: `v1.1.0`: https://github.com/tadfisher/pass-otp/releases

Comment (Mic92) [reply pkgs/tools/security/pass-otp/default.nix]: Version should be also part of the name.

Comment (tadfisher) [reply pkgs/tools/security/pass-otp/default.nix]: Done, thanks!

**Comment by Mic92** [comment]
@GrahamcOfBorg build pass-otp

**Comment by 4z3** [comment]
How is this extension supposed to work?

https://github.com/tadfisher/pass-otp says
```nix
{
  environment.systemPackages = [ pkgs.pass-otp ];
}
```

But just adding that line doesn't work (for me) because that won't make the `pass` executable available. If I add both, `pass` and `pass-otp` to `environment.systemPackages`, then `pass` won't find the extension (because it's not installed in pass's `SYSTEM_EXTENSION_DIR`, i.e. `${pkgs.pass}/lib/password-store/extensions`.

Maybe `pass-otp` should be somehow integrated into `pass` like `import`, `update`, and `tomb` are in https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/security/pass/default.nix. I did it with following hack to achieve that:

```diff
--- a/pkgs/tools/security/pass-otp/default.nix
+++ b/pkgs/tools/security/pass-otp/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, pass, fetchFromGitHub, oathToolkit }:
+{ stdenv, fetchFromGitHub, oathToolkit }:
 stdenv.mkDerivation rec {
   name = "pass-otp-${version}";
   version = "1.1.0";
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
     sha256 = "1cgj4zc8fq88n3h6c0vkv9i5al785mdprpgpbv5m22dz9p1wqvbb";
   };
 
-  buildInputs = [ pass oathToolkit ];
+  buildInputs = [ oathToolkit ];
 
   patchPhase = ''
     sed -i -e 's|OATH=\$(which oathtool)|OATH=${oathToolkit}/bin/oathtool|' otp.bash
diff --git a/tv/5pkgs/simple/pass/default.nix b/tv/5pkgs/simple/pass/default.nix
index ad2afa8e..3b692808 100644
--- a/pkgs/tools/security/pass/default.nix
+++ b/pkgs/tools/security/pass/default.nix
@@ -2,6 +2,8 @@
 , coreutils, gnused, getopt, git, tree, gnupg, which, procps, qrencode
 , makeWrapper
 
+, pass-otp
+
 , xclip ? null, xdotool ? null, dmenu ? null
 , x11Support ? !stdenv.isDarwin
 , tombPluginSupport ? false, tomb
@@ -56,6 +58,14 @@ in stdenv.mkDerivation rec {
       popd
     '') plugins)}
 
+    ln -s \
+        ${pass-otp}/lib/password-store/extensions/otp.bash \
+        $out/lib/password-store/extensions/
+
+    ln -s \
+        ${pass-otp}/share/man/man1/pass-otp.1.gz \
+        $out/share/man/man1/
+
     # Install Emacs Mode. NOTE: We can't install the necessary
     # dependencies (s.el and f.el) here. The user has to do this
     # himself.
```

Alternatively `PASSWORD_STORE_EXTENSIONS_DIR` could be used.

(This question should probably be an issue at https://github.com/tadfisher/pass-otp, but then the above diff wouldn't make sense.)

**Comment by tadfisher** [comment]
@4z3 see https://github.com/NixOS/nixpkgs/pull/39121

**Pull request merged**